bus-mapping/src/evm/opcodes.rs

```diff
@@ -466,6 +467,9 @@ pub fn gen_begin_tx_ops(
     let mut exec_step = state.new_begin_tx_step();
     let call = state.call()?.clone();
 
+    // Add 3 RW read operations for transaction L1 fee.
+    gen_tx_l1_fee_ops(state, &mut exec_step);
```

Comment:
If the l1_fee relatied ops is not put inside (at the beginning of) of `new_begin_tx_step`, I guess it should be put before that rather than after. Since the 'enough balance' check require l1 fee being calculated so the read_op for l1 fee should occur before that

Reply:
It seems that [new_begin_tx_step](https://github.com/scroll-tech/zkevm-circuits/blob/develop/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L64) only constructs ExecStep.

`gen_tx_l1_fee_ops` is added at the beginning of `gen_begin_tx_ops` now.